
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>ChIP-seq analysis with R/Bioconductor - The Cat Way</title>
  <meta name="author" content="Itoshi NIKAIDO">

  
  <meta name="description" content="ChIP-seq Analysis With R/Bioconductor Aug 13th, 2012 RとBioconductorでNGS解析: 3限 ChIP-seq データ解析 ToDo はじめに この文章は統合データベース講習会：AJACSみちのく2「 &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://cat.hackingisbelieving.org//lecture/tohoku-u/NGS-R-Bioconductor-3rd.html">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="The Cat Way" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-50876-8']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


  <!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>
</head>

<body   >
  <header role="banner"><hgroup>
  <h1><a href="/">The Cat Way</a></h1>
  
    <h2>Better be the head of a cat than the tail of a lion.</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:cat.hackingisbelieving.org/" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/lecture/">Lecture</a></li>    
    <li><a href="/links/">Links</a></li>
  <li><a href="/about/">About</a></li>  
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article role="article">
  
  <header>
    <h1 class="entry-title">ChIP-seq Analysis With R/Bioconductor</h1>
    <p class="meta">








  


<time datetime="2012-08-13T18:00:00+09:00" pubdate data-updated="true">Aug 13<span>th</span>, 2012</time></p>
  </header>
  
  <p>RとBioconductorでNGS解析: 3限 ChIP-seq データ解析</p>

<h2>ToDo</h2>

<h2>はじめに</h2>

<p>この文章は統合データベース講習会：AJACSみちのく2「RとBioconductorを使ったNGS解析」3限目「ChIP-seq データ解析の基礎」の講義資料です。</p>

<p>この文章の著作権は二階堂愛にあります。ファイルのダウンロード、印刷、複製、大量の印刷は自由におこなってよいです。企業、アカデミアに関わらず講義や勉強会で配布してもよいです。ただし販売したり営利目的の集まりで使用してはいけません。ここで許可した行為について二階堂愛に連絡や報告する必要はありません。常に最新版を配布したいので<strong>ネット上での再配布や転載は禁止</strong>します。ネット上でのリンクはご自由にどうぞ。内容についての問い合わせはお気軽にメールしてください。</p>

<h2>対象</h2>

<p>ここでは、Rの基礎知識がある人を対象に、RとBioconductor を使った ChIP-seq データ解析について説明します。基本的な Unix コマンドが利用できることが前提です。</p>

<h2>準備</h2>

<h2>学習範囲</h2>

<p>ChIP-seq を取り扱う。Peak解析については、Rではなく、ほかのオープンソースプログラムを利用して解析を行う。Rと Bioconductor ではその後の高次解析から論文の図の作成に至るまでを紹介する。</p>

<h2>ChIP-seq</h2>

<p>illumina HiSeq は、塩基とその精度を表す quality value のセットで出力します。これは fastq file と呼ばれており、すべての解析のスタートになります。</p>

<h3>マッピング</h3>

<p>まずこのファイルに含まれるシーケンスリードをリファレンスゲノムにマッピングすることで、どのゲノム領域由来のリードだったのかを調べます。ここでは、代表的なマッピングソフトである bowtie を用いて、マッピングを行います。</p>

<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>bowtie -t -p 8 -n 3 -m 1 -a --best --strata --sam mm9 -q Oct4.fastq  &gt; Oct4.bowtie.sam
</span><span class='line'>bowtie -t -p 8 -n 3 -m 1 -a --best --strata --sam mm9 -q Sox2.fastq  &gt; Sox2.bowtie.sam
</span><span class='line'>bowtie -t -p 8 -n 3 -m 1 -a --best --strata --sam mm9 -q Nanog.fastq &gt; Nanog.bowtie.sam
</span><span class='line'>bowtie -t -p 8 -n 3 -m 1 -a --best --strata --sam mm9 -q GFP.fastq   &gt; GFP.bowtie.sam</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>samtools view -bS Oct4.bowtie.sam &gt; Oct4.bowtie.bam
</span><span class='line'>samtools sort       Oct4.bowtie.bam Oct4.bowtie.sort
</span><span class='line'>samtools index      Oct4.bowtie.sort.bam
</span><span class='line'>rm Oct4.bowtie.bam</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>macs2 -t results/bowtie/Oct4/Oct4.bowtie.sort.rmRepeat.bam -c results/ bowtie/GFP/GFP.bowtie.sort.rmRepeat.bam -f BAM -g mm -n Oct4 -B -q 0.01</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class=''><span class='line'> source("Oct4_model.r")</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>library("ChIPpeakAnno")
</span><span class='line'>
</span><span class='line'>oct4.df  &lt;- read.table("results/macs2/Oct4_peaks.bed",  header = FALSE)
</span><span class='line'>sox2.df  &lt;- read.table("results/macs2/Sox2_peaks.bed",  header = FALSE)
</span><span class='line'>nanog.df &lt;- read.table("results/macs2/Nanog_peaks.bed", header = FALSE)
</span><span class='line'>
</span><span class='line'>
</span><span class='line'>oct4.gr &lt;- BED2RangedData(oct4.df, header = FALSE)
</span><span class='line'>sox2.gr &lt;- BED2RangedData(sox2.df, header = FALSE)
</span><span class='line'>nanog.gr &lt;- BED2RangedData(oct4.df, header = FALSE)</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>pie(table(as.data.frame(sox2.anno)$insideFeature))
</span><span class='line'>pie(table(as.data.frame(oct4.anno)$insideFeature))
</span><span class='line'>pie(table(as.data.frame(nanog.anno)$insideFeature))</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>library(org.Mm.eg.db)
</span><span class='line'>
</span><span class='line'>oct4.go &lt;- getEnrichedGO(
</span><span class='line'>  oct4.anno,
</span><span class='line'>  orgAnn    = "org.Mm.eg.db",
</span><span class='line'>  maxP      = 0.01,
</span><span class='line'>  multiAdj  = TRUE,
</span><span class='line'>  minGOterm = 10,
</span><span class='line'>  multiAdjMethod = "BH"
</span><span class='line'>)
</span><span class='line'>
</span><span class='line'>oct4.bp.goterm &lt;- unique(oct4.go$bp[order(oct4.go$bp[,10]), c(2,10)])
</span><span class='line'>oct4.cc.goterm &lt;- unique(oct4.go$cc[order(oct4.go$cc[,10]), c(2,10)])
</span><span class='line'>oct4.mf.goterm &lt;- unique(oct4.go$mf[order(oct4.go$mf[,10]), c(2,10)])
</span><span class='line'>head(oct4.mf.goterm)</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>library("BSgenome.Mmusculus.UCSC.mm9")
</span><span class='line'>
</span><span class='line'>oct4.peaksWithSeqs &lt;- getAllPeakSequence(
</span><span class='line'>  oct4.gr,
</span><span class='line'>  upstream   = 0,
</span><span class='line'>  downstream = 0,
</span><span class='line'>  genome = Mmusculus
</span><span class='line'>)
</span><span class='line'>
</span><span class='line'>write2FASTA(oct4.peaksWithSeqs, file="results/3rd/oct4.peaksWithSeqs.fa")</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>library("rGADEM")
</span><span class='line'>
</span><span class='line'>oct4.seqs  &lt;- read.DNAStringSet("results/3rd/oct4.peaksWithSeqs.fa", "fasta")
</span><span class='line'>oct4.motif &lt;- GADEM(oct4.seqs, verbose=1, genome = Mmusculus)</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>library("MotIV")
</span><span class='line'>
</span><span class='line'>## prep. database
</span><span class='line'>path &lt;- system.file(package = "MotIV")
</span><span class='line'>jaspar        &lt;- readPWMfile(file.path(path, "extdata/jaspar2010.txt"))
</span><span class='line'>jaspar.scores &lt;- readDBScores(
</span><span class='line'>  file.path(path, "extdata/jaspar2010_PCC_SWU.scores")
</span><span class='line'>)
</span><span class='line'>
</span><span class='line'>## search motifs
</span><span class='line'>oct4.motif.pwms &lt;- getPWM(oct4.motif)
</span><span class='line'>oct4.jaspar &lt;- motifMatch(
</span><span class='line'>  inputPWM = oct4.motif.pwms,
</span><span class='line'>  align = "SWU",
</span><span class='line'>  cc    = "PCC",
</span><span class='line'>  database = jaspar,
</span><span class='line'>  DBscores = jaspar.scores,
</span><span class='line'>  top = 5
</span><span class='line'>)
</span><span class='line'>
</span><span class='line'>## output results with sequence logos
</span><span class='line'>plot(oct4.jaspar, ncol = 2, top = 5, rev = FALSE,
</span><span class='line'>     main = "Oct4 ChIP-seq", bysim = TRUE)</span></code></pre></td></tr></table></div></figure>




<figure class='code'><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
</pre></td><td class='code'><pre><code class=''><span class='line'>quga.oct4.sox2  &lt;- qugacomp(oct4.bin500,  sox2.bin500)
</span><span class='line'>quga.oct4.nanog &lt;- qugacomp(oct4.bin500, nanog.bin500)
</span><span class='line'>quga.sox2.nanog &lt;- qugacomp(sox2.bin500, nanog.bin500)
</span><span class='line'>
</span><span class='line'>num &lt;- 3
</span><span class='line'>mat.cor &lt;- matrix(0, nrow=num, ncol=num)
</span><span class='line'>rownames(mat.cor) &lt;- c("Oct4", "Sox2", "Nanog")
</span><span class='line'>colnames(mat.cor) &lt;- c("Oct4", "Sox2", "Nanog")
</span><span class='line'>
</span><span class='line'>diag(mat.cor) &lt;- rep(1, num)
</span><span class='line'>
</span><span class='line'>mat.cor[1,2] &lt;- mat.cor[2,1] &lt;- pearsonCoef(quga.oct4.sox2)
</span><span class='line'>mat.cor[1,3] &lt;- mat.cor[3,1] &lt;- pearsonCoef(quga.oct4.nanog)
</span><span class='line'>mat.cor[2,3] &lt;- mat.cor[3,2] &lt;- pearsonCoef(quga.sox2.nanog)
</span><span class='line'>
</span><span class='line'>mat.cor.max &lt;- max( mat.cor[upper.tri(mat.cor, diag=F)] )
</span><span class='line'>mat.cor.min &lt;- min( mat.cor[upper.tri(mat.cor, diag=F)] )
</span><span class='line'>
</span><span class='line'>corrplot(
</span><span class='line'>  mat.cor, method="circle", type="full", diag=FALSE, outline=FALSE,
</span><span class='line'>  addcolorlabel="bottom", cl.lim=c(mat.cor.min, mat.cor.max), cl.ratio=0.2
</span><span class='line'>)</span></code></pre></td></tr></table></div></figure>


  
    <footer>
      <p class="meta">
        
        








  


<time datetime="2012-08-13T18:00:00+09:00" pubdate data-updated="true">Aug 13<span>th</span>, 2012</time>
        
      </p>
      
        <div class="sharing">
  
  <a href="http://twitter.com/share" class="twitter-share-button" data-url="http://cat.hackingisbelieving.org//lecture/tohoku-u/NGS-R-Bioconductor-3rd.html" data-via="dritoshi" data-counturl="http://cat.hackingisbelieving.org//lecture/tohoku-u/NGS-R-Bioconductor-3rd.html" >Tweet</a>
  
  
  <div class="g-plusone" data-size="medium"></div>
  
  
    <div class="fb-like" data-send="true" data-width="450" data-show-faces="false"></div>
  
</div>

      
    </footer>
  
</article>

  <section>
    <h1>Comments</h1>
    <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
  </section>

</div>

<aside class="sidebar">
  
    <section>
  <h1>About</h1>
  <p>二階堂愛(にかいどういとし♂)のメモ帳。詳しくは<a href="about/">こちら</a>をどうぞ。</P>
</section>
<section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2012/08/16/how-to-change-key-bindings-in-the-rstudio-console/">How to change key bindings in the RStudio console</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/08/14/how-to-install-octopress-on-your-github-pages/">How to install Octopress on your Github Pages</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/08/14/how-to-write-a-new-entry/">How to write a new entry</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/08/13/r-markdown-with-octopress/">R Markdown with Octopress on your GitHub</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/08/13/NGS-R-Bioconductor-1st/">NGS analysis with R and Bioconductor</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/03/31/sampling-from-probability-density-function/">Sampling from Probability Density Function</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/03/31/mcmc-tutorials-on-web/">MCMC Tutorials on Web</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/dritoshi">@dritoshi</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'dritoshi',
            count: 3,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>


<section>
  <h1>Latest Tweets</h1>
  <ul id="tweets">
    <li class="loading">Status updating...</li>
  </ul>
  <script type="text/javascript">
    $.domReady(function(){
      getTwitterFeed("dritoshi", 3, false);
    });
  </script>
  <script src="/javascripts/twitter.js" type="text/javascript"> </script>
  
    <a href="http://twitter.com/dritoshi" class="twitter-follow-button" data-show-count="true">Follow @dritoshi</a>
  
</section>


<section>
  <h1>On Delicious</h1>
  <div id="delicious"></div>
  <script type="text/javascript" src="http://feeds.delicious.com/v2/json/dritoshi?count=3&amp;sort=date&amp;callback=renderDeliciousLinks"></script>
  <p><a href="http://delicious.com/dritoshi">My Delicious Bookmarks &raquo;</a></p>
</section>


<section class="googleplus">
  <h1>
    <a href="https://plus.google.com/116140930812537145672?rel=author">
      <img src="http://www.google.com/images/icons/ui/gprofile_button-32.png" width="32" height="32">
      Google+
    </a>
  </h1>
</section>



  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2012 - Itoshi NIKAIDO -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'catway';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://cat.hackingisbelieving.org//lecture/tohoku-u/NGS-R-Bioconductor-3rd.html';
        var disqus_url = 'http://cat.hackingisbelieving.org//lecture/tohoku-u/NGS-R-Bioconductor-3rd.html';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = 'http://' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>



<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
